internal/olm/operator/registry/index_image.go

```diff
@@ -146,8 +478,41 @@ func (c IndexImageCatalogCreator) UpdateCatalog(ctx context.Context, cs *v1alpha
 			// if no annotations exist and image reference is empty, error out
 			return errors.New("cannot upgrade: no catalog image reference exists in catalog source spec or annotations")
 		}
+
 		// if no annotations exist and image reference exists, set it to index image
 		c.IndexImage = cs.Spec.Image
+
+		// check if index image adopts File-Based Catalog or SQLite index image format
+		isFBCImage, err := isFBC(ctx, c.IndexImage)
```

Comment:
The reason I have it like this is because the value of `c.IndexImage` will get update right before it, so we want to make sure that the `isFBC` function is called on the updated index image.

Since I make these calls inside the if/else, it won't get called twice. It'll only get called once based on which portion of that block is executed.